src/main/java/com/expensetracker/service/impl/ExpensePeriodServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package com.expensetracker.service.impl;
+
+import com.expensetracker.model.*;
+import com.expensetracker.repository.*;
+import com.expensetracker.service.ExpensePeriodService;
+import com.expensetracker.service.ExpenseService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.BeanUtils;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class ExpensePeriodServiceImpl implements ExpensePeriodService {
+
+    private final ExpensePeriodRepository expensePeriodRepository;
+
+    private final ExpenseService expenseService;
+
+    private final CategoryRepository categoryRepository;
+
+    private final WalletRepository walletRepository;
+
+    private final SubcategoryRepository subcategoryRepository;
+
+    @Override
+    public ExpensePeriod createNewExpensePeriod(ExpensePeriod expensePeriod) {
+        Category category = categoryRepository.findByName(expensePeriod.getCategory().getName());
+        expensePeriod.setCategory(category);
+
+        if (expensePeriod.getSubcategory().getName() == null) {
+            expensePeriod.setSubcategory((Subcategory) null);
+        } else {
+            Subcategory subcategory = subcategoryRepository.findByName(expensePeriod.getSubcategory().getName());
+            expensePeriod.setSubcategory(subcategory);
+        }
+
+        Wallet wallet = walletRepository.findByName(expensePeriod.getWallet().getName());
+        expensePeriod.setWallet(wallet);
+
+        expensePeriod.setNextPayment(expensePeriod.getStartDate());
+
+        createExpenses(expensePeriod);
+
+        return expensePeriodRepository.save(expensePeriod);
+    }
+
+    @Override
+    public void createExpenses(ExpensePeriod expensePeriod) {
+        while (expensePeriod.getNextPayment() != null && !expensePeriod.getNextPayment().isAfter(LocalDate.now())) {
+            expenseService.saveExpenseFromPeriod(expensePeriod);
+
+            setExpenseNextPayment(expensePeriod);
+            expensePeriodRepository.save(expensePeriod);
+        }
+    }
+
+    private void setExpenseNextPayment(ExpensePeriod expensePeriod) {
+        LocalDate nextPayment = null;
+
+        switch (expensePeriod.getPeriodType()) {
+            case "YEAR":
```

Comment:
You could also use the enhanced Switch/Case from Java 17 so you don't need to use the break.